src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexWriteOperation.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Corax;
+using Raven.Client.Documents.Indexes;
+using Raven.Server.Exceptions;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Logging;
+using Voron;
+using Voron.Impl;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Corax
+{
+    public class CoraxIndexWriteOperation : IndexWriteOperationBase
+    {
+        private readonly IndexWriter _indexWriter;
+        private readonly CoraxDocumentConverter _converter;
+        private readonly Dictionary<Slice, int> _knownFields;
+        private int _entriesCount = 0;
+
+        public CoraxIndexWriteOperation(Index index, Transaction writeTransaction, CoraxDocumentConverter converter, Logger logger) : base(index, logger)
+        {
+            _converter = converter;
+            _knownFields = _converter.GetKnownFields();
+            try
+            { 
+                _indexWriter = new IndexWriter(writeTransaction);
+            }
+            catch (Exception e) when (e.IsOutOfMemory())
+            {
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw new IndexWriteException(e);
+            }
+        }
+
+        public override void Dispose()
+        {
+            _indexWriter?.Dispose();
+        }
+
+        public override void Commit(IndexingStatsScope stats)
+        {
+            if (_indexWriter != null)
+            {
+                using (stats.For(IndexingOperation.Corax.Commit))
+                    _indexWriter.Commit();
+            }
+        }
+
+        public override void IndexDocument(LazyStringValue key, LazyStringValue sourceDocumentId, object document, IndexingStatsScope stats, JsonOperationContext indexContext)
+        {
+            EnsureValidStats(stats);
+            _entriesCount++;
+            Span<byte> data;
+            string id;
+
+            using (Stats.ConvertStats.Start())
+                data = _converter.InsertDocumentFields(key, sourceDocumentId, document, indexContext, out id);
+
+            _indexWriter.Index(id, data, _knownFields);
+            stats.RecordIndexingOutput();
+
+        }
+
+        public override int EntriesCount() => _entriesCount;
+
+        public override (long RamSizeInBytes, long FilesAllocationsInBytes) GetAllocations()
+        {
+            return (1024 * 1024, 1024 * 1024);
```

Comment:
add `TODO`

Reply:
done

Reply:
@redknightlois What should we put here?

Reply:
For encrypted storage / compressed - we have some details on the transaction.
Otherwise, we can look at the number of modified pages, I think. Or maybe include the size of allocations on the allocator